src/openmrs.ts

```diff
@@ -11,9 +11,29 @@ function setupOpenMRS() {
   setupEncounterRequestInterceptors();
   setupOfflineEncounterSync();
 
+  // TODO: This event must be replaced with a better one, e.g. sth like "subscribePrecache".
```

Comment:
Should we already introduce this?

Reply:
Generally yes, as soon as possible. But I think we need to brainstorm a little bit about how the event should look like and whether it will be integrated with, e.g., the patient list or not.

Reply:
Would it only be required to run once?

Since we don't know "when" we are going offline we should call it any time we are online, i.e., in the beginning (if online) and then later if we toggled from offline to online. Presumably, we also should have a threshold time (e.g., 5 minutes) to avoid calling it too often. So the minimum "outdated" data when toggling (or restarting the application) would be X (5) minutes.

Reply:
That would be easy-ish enough to do right now. I'll look into it!

Reply:
OK let's wait then with a merge :) ! Thanks!

Reply:
Alright, the new event is in now!